File: csharp/HearthstoneReplays/Parser/Handlers/SendOptionHandler.cs
#region

using System;
using HearthstoneReplays.Parser.ReplayData;
using HearthstoneReplays.Parser.ReplayData.Meta.Options;

#endregion

namespace HearthstoneReplays.Parser.Handlers
{
	public class SendOptionHandler
	{
		public static void Handle(string timestamp, string data, ParserState state)
		{
			data = data.Trim();
			var match = Regexes.SendOptionRegex.Match(data);
			if(match.Success)
			{
				var option = match.Groups[1].Value;
				var suboption = match.Groups[2].Value;
				var target = match.Groups[3].Value;
				var position = match.Groups[4].Value;
				var sendOption = new SendOption
				{
					OptionIndex = int.Parse(option),
					Position = int.Parse(position),
					SubOption = int.Parse(suboption),
					Target = int.Parse(target)
				};
				if(state.Node.Type == typeof(Game))
					((Game)state.Node.Object).Data.Add(sendOption);
				else
					throw new Exception("Invalid node " + state.Node.Type + " -- " + data);
			}
		}
	}
}

File: java/src/main/java/info/hearthsim/hsreplay/parser/handlers/SendOptionHandler.java
package info.hearthsim.hsreplay.parser.handlers;

import info.hearthsim.hsreplay.parser.ParserState;
import info.hearthsim.hsreplay.parser.Regexes;
import info.hearthsim.hsreplay.parser.replaydata.Game;
import info.hearthsim.hsreplay.parser.replaydata.meta.options.SendOption;

import java.util.regex.Matcher;

public class SendOptionHandler {

	public static void handle(String timestamp, String data, ParserState state) throws Exception {
		data = data.trim();
		Matcher match = Regexes.SendOptionRegex.matcher(data);
		if (match.matches()) {
			String option = match.group(1);
			String suboption = match.group(2);
			String target = match.group(3);
			String position = match.group(4);

			SendOption sendOption = SendOption.builder().optionIndex(Integer.parseInt(option))
					.position(Integer.parseInt(position)).subOption(Integer.parseInt(suboption))
					.target(Integer.parseInt(target)).build();

			if (state.getNode().getType().isAssignableFrom(Game.class))
				((Game) state.getNode().getObject()).getData().add(sendOption);
			else
				throw new Exception("Invalid node " + state.getNode().getType());
		}
	}
}
